bql/lexer/lexer.go

```diff
@@ -603,6 +616,22 @@ func lexNode(l *lexer) stateFn {
 	return lexSpace
 }
 
+// lexBlankNode tries to lex a blank node out of the input
+func lexBlankNode(l *lexer) stateFn {
+	if r := l.next(); r != colon {
+		l.emitError("blank node should start with _:")
+		return nil
+	}
+	for {
+		if r := l.next(); !unicode.IsLetter(r) && !unicode.IsDigit(r) && r != rune('_') || r == eof {
```

Comment:
Do we want to force the first rune to be a letter? right now this `_:_` or `_:1` would be valid blank node IDs?